aineko/core/dataset.py

```diff
@@ -204,3 +180,72 @@ def exists(self, *args: T, **kwargs: T) -> bool:
         This method should return True if the dataset exists, otherwise False.
         """
         raise NotImplementedError
+
+    @abc.abstractmethod
+    def setup_test_mode(
+        self,
+        source_node: str,
+        source_pipeline: str,
+        input_values: Optional[List[dict]] = None,
+    ) -> None:
+        """Subclass implementation to set up the dataset for testing.
+
+        Nodes have the ability to run in test mode, which allows them to run
+        without setting up the actual dataset storage layer. All dataset
+        implementations must implement this method. A dataset in test mode
+        should never interact with the real storage layer. Instead, it should
+        use the class attributes as the storage layer:
+
+        * `_input_values` for input values
+        * `_output_values` for output values
+
+        Args:
+            source_node: The name of the source node.
+            source_pipeline: The name of the source pipeline.
+            input_values: A list of input values.
+        """
+        raise NotImplementedError
```

Comment:
I'm thinking it might be good to set up a very simple default behavior for this so people can create a dataset quickly without having to go through too many steps. Something like setting up a list or dictionary that stores data.

Reply:
Yes that could be good, but I'm not sure how feasible it is (at this point) to create a good implementation that would work for all future datasets. I would rather postpone that until we have a few more datasets, or have enough time to research and design it.

If we really want to do it now, we could do a simple implementation like the KafkaDataset with an internal list. But we should decide what to do about the message wrapper:
* do we keep it in it's current form and make a pydantic model for it?
* do we want to remove some fields?
* ...

```python
{
    "timestamp": datetime.datetime.now().strftime(
        AINEKO_CONFIG.get("MSG_TIMESTAMP_FORMAT")
    ),
    "message": message,
    "source_node": self.source_node,
    "source_pipeline": self.source_pipeline,
}
```

Reply:
ok i'm fine with us doing that a little later